src/components/street-generated-stencil.js

```diff
@@ -0,0 +1,176 @@
+    },
+    cycleOffset: {
+      // z (inbound/outbound) offset as a fraction of spacing value
+      default: 0.5, // this is used to place different models at different z-levels with the same spacing value
+      type: 'number'
+    },
+    facing: {
+      default: 0, // this is a Y Rotation value in degrees -- UI could offer a dropdown with options for 0, 90, 180, 270
+      type: 'number'
+    },
+    randomFacing: {
+      // if true, facing is ignored and a random Y Rotation is applied to each clone
+      default: false,
+      type: 'boolean'
+    },
+    stencilHeight: {
+      default: 0,
+      type: 'number'
+    },
+    direction: {
+      // specifying inbound/outbound directions will overwrite facing/randomFacing
+      type: 'string',
+      oneOf: ['none', 'inbound', 'outbound']
+    }
+    // seed: {  // seed not yet supported
+    //   default: 0,
+    //   type: 'number'
+    // }
+  },
+  init: function () {
+    this.createdEntities = [];
+  },
+  remove: function () {
+    this.createdEntities.forEach((entity) => entity.remove());
+  },
+  update: function (oldData) {
+    const data = this.data;
+    if (AFRAME.utils.deepEqual(oldData, data)) return;
+
+    // Clean up old entities
+    this.createdEntities.forEach((entity) => entity.remove());
+    this.createdEntities = [];
+
+    // Use either stencils array or single model
+    let stencilsToUse = data.stencils.length > 0 ? data.stencils : [data.model];
+
+    // Reverse stencil order if inbound
+    if (data.direction === 'inbound') {
+      stencilsToUse = stencilsToUse.slice().reverse();
+    }
+
+    // Ensure minimum spacing
+    this.correctedSpacing = Math.max(1, data.spacing);
+
+    // Calculate number of stencil groups that can fit in the length
+    const numGroups = Math.floor(data.length / this.correctedSpacing);
+
+    // Create stencil groups along the street
+    for (let groupIndex = 0; groupIndex < numGroups; groupIndex++) {
+      const groupPosition =
+        data.length / 2 -
+        (groupIndex + data.cycleOffset) * this.correctedSpacing;
+
+      // Create each stencil within the group
+      stencilsToUse.forEach((stencilName, stencilIndex) => {
+        const clone = document.createElement('a-entity');
+        clone.setAttribute('mixin', stencilName);
+
+        // Calculate stencil position within group
+        const stencilOffset =
+          (stencilIndex - (stencilsToUse.length - 1) / 2) * data.padding;
+
+        // Set position with group position and stencil offset
+        clone.setAttribute('position', {
+          x: data.positionX,
+          y: data.positionY,
+          z: groupPosition + stencilOffset
+        });
+
+        // Handle stencil height if specified
+        if (data.stencilHeight > 0) {
+          clone.addEventListener('loaded', (evt) => {
+            evt.target.setAttribute('geometry', 'height', data.stencilHeight);
+            evt.target.setAttribute('atlas-uvs', 'forceRefresh', true);
```

Comment:
"core:schema:warn Unknown property `forceRefresh` for component/system `atlas-uvs`. 
`evt.target.components['atlas-uvs'].update()`
may be better here?

Reply:
Is this the same hack with the non existing property here? https://github.com/3DStreet/3dstreet/blob/107d79786f9a477b5eb15cc2281eb6d799cb0110/src/aframe-streetmix-parsers.js#L25-L26

Reply:
e4a0cc2d553ce176fd412a851a777a7d4e9f9021